bamt/networks/base.py

```diff
@@ -458,11 +459,15 @@ def fit_parameters(self, data: pd.DataFrame, dropna: bool = True):
 
         def worker(node):
             return node.fit_parameters(data)
-
-        pool = ThreadPoolExecutor(3)
-        for node in self.nodes:
-            future = pool.submit(worker, node)
-            self.distributions[node.name] = future.result()
+          
+        results = Parallel(n_jobs=-1)(delayed(worker)(node) for node in self.nodes)
```

Comment:
В идеале, конечно, хотелось бы, чтобы n_jobs был не захардкоженный на -1, а передавался в метод при вызове, чтобы пользователь сам мог задать количество потоков, которое он может выделить на эту задачу, к примеру:

~~~python
bn.fit_parameters(..., n_jobs = n)
~~~

Для этого можно просто отредактировать входы метода:

~~~python
def fit_parameters(self, data: pd.DataFrame, dropna: bool = True, n_jobs: int = -1):
~~~

а в этом месте сделать так
```suggestion
        results = Parallel(n_jobs=n_jobs)(delayed(worker)(node) for node in self.nodes)
```